Tests/swift-sdk-swift-tests/IterableNotificationResponseTests.swift

```diff
@@ -50,14 +63,23 @@ class IterableNotificationResponseTests: XCTestCase {
 
         let response = MockNotificationResponse(userInfo: userInfo, actionIdentifier: UNNotificationDefaultActionIdentifier)
         let pushTracker = MockPushTracker()
-        let customActionDelegate = MockCustomActionDelegate(returnValue: true)
-        
-        let actionRunner = IterableActionRunner(urlDelegate: nil, customActionDelegate: customActionDelegate, urlOpener: MockUrlOpener())
-        let appIntegration = IterableAppIntegrationInternal(tracker: pushTracker, actionRunner: actionRunner, versionInfo: MockVersionInfo(version: 10))
+        let promise = Promise<String, Error>()
```

Comment:
Why are promises necessary here? Wouldn't it be enough to just pass a mock `customActionHandler`?

Reply:
Not really. I will change to expectations.